src/apiserver/api_service_impl.h

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SRC_APISERVER_API_SERVICE_IMPL_H_
+#define SRC_APISERVER_API_SERVICE_IMPL_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "apiserver/interface_provider.h"
+#include "json2pb/rapidjson.h"  // rapidjson's DOM-style API
+#include "proto/http.pb.h"
+#include "sdk/sql_cluster_router.h"
+
+namespace fedb {
+namespace http {
+
+using butil::rapidjson::Document;
+using butil::rapidjson::StringBuffer;
+using butil::rapidjson::Writer;
+
+struct Column;
+typedef std::vector<Column> SimpleSchema;
+
+class APIServiceImpl : public APIService {
+ public:
+    APIServiceImpl() = default;
+    ~APIServiceImpl() override;
+    bool Init(const sdk::ClusterOptions& options);
+    void Process(google::protobuf::RpcController* cntl_base, const HttpRequest*, HttpResponse*,
+                 google::protobuf::Closure* done) override;
+
+ private:
+    static bool Json2SQLRequestRow(const butil::rapidjson::Value& input, const butil::rapidjson::Value& common_cols_v,
+                                   std::shared_ptr<fedb::sdk::SQLRequestRow> row);
+    static bool AppendJsonValue(const butil::rapidjson::Value& v, hybridse::sdk::DataType type,
+                                std::shared_ptr<fedb::sdk::SQLRequestRow> row);
+    void RegisterPut();
+    void RegisterExecSP();
+    void RegisterGetSP();
+
+    static SimpleSchema TransToSimpleSchema(const hybridse::sdk::Schema* schema) {
+        SimpleSchema ss;
+        for (int i = 0; i < schema->GetColumnCnt(); ++i) {
+            ss.emplace_back(schema->GetColumnName(i), schema->GetColumnType(i), schema->IsConstant(i),
+                            schema->IsColumnNotNull(i));
+        }
+        return std::move(ss);
```

Comment:
I think `std::move` is not neccessary, there is copy elision

Reply:
fixed